src/trodes_to_nwb/convert_statescript.py

```diff
@@ -0,0 +1,1254 @@
+        bitwise masks representing the state of DIO pins.
+        Example: ``1817158 128 512``
+
+    ``ts_str_int`` : `<timestamp_int> <str> <int>`
+        Represents timestamp, a string label, and an integer value. Frequently
+        used for user-defined messages logging DIO pin state changes (e.g., pin name and state).
+        Example: ``8386500 DOWN 3``
+
+    ``ts_str_eq_int`` : `<timestamp_int> <str> = <int>`
+        Represents timestamp and a named integer variable assignment, useful for
+        tracking counters or state variables within the StateScript.
+        Example: ``3610855 totRewards = 70``
+
+    ``ts_str`` : `<timestamp_int> <str...>`
+        Represents timestamp followed by one or more space-separated strings.
+        Commonly used for logging event markers or descriptive text messages.
+        Example: ``1678886401 LOCKEND``
+
+    ``comment_or_empty`` : Lines starting with `#` or completely empty lines.
+        Lines starting with '#' are treated as comments. Empty lines may also occur.
+        These are typically ignored during data extraction.
+        Example: ``# Starting new trial block``
+
+    ``unknown`` : Lines that do not conform to the patterns listed above.
+        These might include initial header lines, formatting variations, or unexpected entries.
+        Example: ``initiated``
+
+Component Definitions:
+    - ``<timestamp_int>``: 64-bit integer; milliseconds since session start (Trodes timestamp).
+    - ``<int>``: Integer value; often used as a bitwise mask for DIO pin states.
+    - ``<str>``: String value; can represent an event name, variable name, message component, etc.
+    - ``<str...>``: Denotes one or more space-separated strings.
+
+"""
+
+import pathlib
+from typing import Any, Dict, List, Optional, Type, TypeVar, Union
+
+import numpy as np
+import pandas as pd
+
+from .convert_dios import _get_channel_name_map as _get_dio_channel_name_map
+
+T_StateScriptLogProcessor = TypeVar(
+    "T_StateScriptLogProcessor", bound="StateScriptLogProcessor"
+)
+
+
+def _parse_int(s: str) -> Optional[int]:
+    """Attempts to parse a string as an integer.
+
+    Parameters
+    ----------
+    s : str
+        Input string.
+
+    Returns
+    -------
+    Optional[int]
+        The parsed integer, or None if parsing fails.
+    """
+    try:
+        return int(s)
+    except ValueError:
+        return None
+
+
+def parse_ts_int_int(parts: list) -> Optional[Dict[str, Any]]:
+    """Parses lines with <timestamp_int> <int> <int> structure.
+
+    This pattern typically represents a timestamp followed by two integer values.
+    These integers are bitwise masks or state values, often used for logging
+    DIO states or other binary values.
+
+    Example:
+        8386500 0 0 -> {'ts': 8386500, 'value1': 0, 'value2': 0}
+        1817158 128 512 -> {'ts': 1817158, 'value1': 128, 'value2': 512}
+        76566 65536 0 -> {'ts': 76566, 'value1': 65536, 'value2': 0}
```

Comment:
```suggestion
        8386500 0 0 -> {'ts': 8386500, 'value1': 0, 'value2': 0}
```

I don't think the extra examples add much